packages/web/pages/pool/[id].tsx

```diff
@@ -13,97 +14,117 @@ import { useTranslation, useWindowSize } from "~/hooks";
+    const router = useRouter();
+    const { chainStore, queriesStore } = useStore();
+    const { chainId } = chainStore.osmosis;
+    const { t } = useTranslation();
+    const { isMobile } = useWindowSize();
+
+    const queryOsmosis = queriesStore.get(chainId).osmosis!;
+
+    const flags = useFeatureFlags();
+
+    const [showTradeModal, setShowTradeModal] = useState(false);
+
+    // eject to pools page if pool does not exist
+    const poolExists =
+      poolId && typeof poolId === "string" && Boolean(poolId)
+        ? queryOsmosis.queryPools.poolExists(poolId)
+        : undefined;
+    useEffect(() => {
+      if (poolExists === false) {
+        router.push("/pools");
+      }
+    }, [poolExists, router]);
+
+    const queryPool = queryOsmosis.queryPools.getPool(poolId);
+
+    useNavBar(
+      useMemo(
+        () => ({
+          title: t("pool.title", { id: poolId ?? "" }),
+          ctas: [
+            { label: t("pool.swap"), onClick: () => setShowTradeModal(true) },
+          ],
+        }),
+        [t, poolId]
+      )
+    );
+
+    useEffect(() => {
+      if (
+        queryPool &&
+        !flags.concentratedLiquidity &&
+        queryPool.type === "concentrated" &&
+        !isMobile
+      ) {
+        router.push(`/pools`);
+      }
+    }, [queryPool, isMobile, flags.concentratedLiquidity, router]);
+
+    return (
+      <>
+        <NextSeo title={t("seo.pool.title", { id: poolId })} />
+        {queryPool && Boolean(poolId) && (
+          <TradeTokens
+            className="md:!p-0"
+            isOpen={showTradeModal}
+            onRequestClose={() => {
+              setShowTradeModal(false);
+            }}
+            sendTokenDenom={queryPool.poolAssetDenoms[0]}
+            outTokenDenom={queryPool.poolAssetDenoms[1]}
+            forceSwapInPoolId={poolId}
+          />
+        )}
+        {!queryPool ? (
+          <div className="mx-auto flex max-w-container flex-col gap-10 py-6 px-6">
+            <SkeletonLoader className="h-[30rem] !rounded-3xl" />
+            <SkeletonLoader className="h-40 !rounded-3xl" />
+            <SkeletonLoader className="h-8 !rounded-xl" />
+            <SkeletonLoader className="h-40 !rounded-3xl" />
+          </div>
+        ) : (
+          <>
+            {flags.concentratedLiquidity &&
+            queryPool?.type === "concentrated" &&
+            !isMobile ? (
+              <ConcentratedLiquidityPool poolId={poolId} />
+            ) : Boolean(queryPool?.sharePool) ? (
+              queryPool && <SharePool poolId={poolId} />
+            ) : queryPool ? (
+              <BasePoolDetails pool={queryPool!.pool} />
+            ) : null}
+          </>
+        )}
+      </>
+    );
+  }
+);
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const { num_pools } = await queryNumPools();
+  let paths: { params: { id: string } }[] = [];
+
+  for (let i = 0; i <= Number(num_pools); i++) {
+    paths.push({ params: { id: String(i + 1) } });
+  }
```

Comment:
Nit: I think a `map` (vs `for`) would be more readable here

Reply:
Done in 2bb4d66